### PR TITLE
feat: quality content corner

### PR DESCRIPTION
**Pull request by Jessuhh**
Added the quality content corner. When a message reaches 5 reactions of a specified emoji (defaults to upvote) it will send the message in a specified channel. The amount of reactions, emoji name and channel to send the message in can be set in the `.env`.
image
image

### CODE REVIEW

File: src/events/UpvoteEvent.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+            // Check if reaction count is above or equal to the minimal emoji count threshold
+            if ((reaction.count ?? 0) < QCC_MIN_EMOJI_COUNT) return;
+
+            // Get the emoji
+            const emoji = reaction.client.emojis.cache.find((emoji) =>
+                emoji.name?.toLowerCase().includes(QCC_EMOJI_NAME),
+            );
+            // Make sure an emoji was found
+            if (!emoji) {
+                logger.fatal("Couldn't find emoji with name", QCC_EMOJI_NAME);
```
Comment (rickklaasboer) [inline src/events/UpvoteEvent.ts:27]: Should probably use error instead of fatal

File: src/events/UpvoteEvent.ts [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+            // Check if channel has been found
+            if (!channel) {
+                logger.fatal('Could not find channel to send message in');
+                return;
+            }
+
+            // Check if channel is a text channel
+            if (!channel.isTextBased()) {
+                logger.fatal('Channel is not a text channel');
```
Comment (rickklaasboer) [inline src/events/UpvoteEvent.ts:47]: Should probably use error instead of fatal

File: src/events/UpvoteEvent.ts [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+            // Check if a message already exists
+            if (message) {
+                // Edit the existing message
+                await message.edit({
+                    content: `${reaction.count} <:upvote:${emoji?.id}> in <#${reaction.message.channelId}>`,
+                });
+            } else {
+                // Create embed
+                const embed: APIEmbed = {
```
Comment (rickklaasboer) [inline src/events/UpvoteEvent.ts:70]: You could rewrite this to make use of the embed builder

Comment (Jessuhh) [reply src/events/UpvoteEvent.ts:70]: You prefer the embed builder? Should be an easy task 👍🏼

**Review by rickklaasboer — APPROVED** [review]

**Pull request merged**